src/docs/sphinx/dev_guide/gpu_porting.rst

```diff
@@ -161,6 +160,199 @@ GPU device, or on both a GPU device and a CPU host. For example::
+or the code will not compile under nvcc.
+
+Do this:
+
+  .. code-block:: cpp
+
+      axom::for_all<ExecSpace>(n, AXOM_LAMBDA(axom::IndexType index) { /* body */});
+
+
+Do NOT do this:
+
+  .. code-block:: cpp
+
+      axom::for_all<ExecSpace>(n, AXOM_LAMBDA(auto index) { /* body */});
+
+
+Data in Axom is often contained in useful containers such as ``axom::Array``.
+Use ``axom::ArrayView`` to access array data from within kernels. Views contain
+a pointer to the data and memory shape information and can be constructed within
+device code to access data arrays. Views are captured by kernels and passed to
+the device code; this copies the pointer and shape information into an object that
+the device code can access. If the view was passed into a method where the ``axom::for_all``
+function is calling a kernel, be sure to pass the view by value so the compiler
+does not capture a host reference to the view, causing the kernel to fail.
+
+Do this:
+
+  .. code-block:: cpp
+
+      template <typename ExecSpace>
+      void doSomething(axom::ArrayView<int> dataView)
+      {
+        axom::for_all<ExecSpace>(dataView.size(), AXOM_LAMBDA(axom::IndexType index)
+        {
+          /* body uses dataView[index] */
+        });
+      }
+
+Do NOT do this:
+
+  .. code-block:: cpp
+
+      template <typename ExecSpace>
+      void doSomething(axom::ArrayView<int> &dataView)
+      {
+        axom::for_all<ExecSpace>(dataView.size(), AXOM_LAMBDA(axom::IndexType index)
+        {
+          /* body uses dataView[index] */
+          /* It will crash on GPU devices because the host reference was
+             captured rather than the object.
+           */
+        });
+      }
+
+If pass by reference is used, create a new view inside the method and then use that
+"device" view in the kernel so the device code uses an object captured by value.
+
+
+Do not call ``axom::for_all`` from class methods that are marked as protected or private.
+The nvcc compiler requires the method containing the kernel to be publicly accessible.
+
+Do this:
+
+  .. code-block:: cpp
+
+      class Algorithm
+      {
+      public:
+        // stuff
+      #if !defined(__CUDACC__)
+      private:
+      #endif
+        void helperMethod()
+        {
+          axom::for_all<ExecSpace>(n, AXOM_LAMBDA(axom::IndexType index) { /* body */});
+        }
+      };
+
+Do NOT do this:
+
+  .. code-block:: cpp
+
+      class Algorithm
+      {
+      public:
+        // stuff
+      private:
+        void helperMethod()
+        {
+          axom::for_all<ExecSpace>(n, AXOM_LAMBDA(axom::IndexType index) { /* body */});
+        }
+      };
+
+When calling a kernel via ``axom::for_all`` from a surrounding lambda function, 
+consider calling ``axom::for_all`` from a class member method instead. The nvcc compiler will
+not compile kernel invokations inside lambda functions. This pattern comes up when an intermediate
```

Comment:
Typo: invokations should be spelled with a c

Reply:
Fixed